modules/dnn/src/layers/resize_layer.cpp

```diff
@@ -45,10 +45,20 @@ class ResizeLayerImpl : public ResizeLayer
                          std::vector<MatShape> &outputs,
                          std::vector<MatShape> &internals) const CV_OVERRIDE
     {
-        CV_Assert_N(inputs.size() == 1, inputs[0].size() == 4);
+        CV_Assert_N(inputs.size() == 1 || inputs.size() == 2, inputs[0].size() == 4);
         outputs.resize(1, inputs[0]);
-        outputs[0][2] = zoomFactorHeight > 0 ? (outputs[0][2] * zoomFactorHeight) : outHeight;
-        outputs[0][3] = zoomFactorWidth > 0 ? (outputs[0][3] * zoomFactorWidth) : outWidth;
+        if (inputs.size() == 1) {
+            outputs[0][2] = zoomFactorHeight > 0 ? (outputs[0][2] * zoomFactorHeight) : outHeight;
+            outputs[0][3] = zoomFactorWidth > 0 ? (outputs[0][3] * zoomFactorWidth) : outWidth;
+        } else {
+            float factor_h = static_cast<float>(inputs[1][2]) / inputs[0][2];
+            float factor_w = static_cast<float>(inputs[1][3]) / inputs[0][3];
+            outputs[0][2] = inputs[1][2];
+            outputs[0][3] = inputs[1][3];
+            if ((interpolation == "linear" && (factor_w != 1.0 || factor_h != 1.0)) ||
+                (interpolation == "nearest" && (factor_w < 1.0 || factor_h < 1.0)))
+                    CV_Error(Error::StsNotImplemented, "Unsupported Resize mode");
```

Comment:
Why they are not supported?

Reply:
Caffe result with these factors is not equal to OpenCV resize